libres/lib/analysis/update.cpp

```diff
@@ -550,10 +550,9 @@ Check whether the current state and config allows the update algorithm
 to be executed
 */
 bool assert_update_viable(const analysis_config_type *analysis_config,
-                          const enkf_fs_type *source_fs,
+                          const state_map_type *source_state_map,
```

Comment:
How come is it possible to replace `enkf_fs_type` with `state_map_type`?

Reply:
Is not a replace perse, the change is done for two reasons. to reduce redundant function invokations and validation function should be in the business of observing not creating or fetching objects.
old implementation only uses `enkf_fs_type` to fetch a `state_map_type`, so if you already have an `enkf_fs_type` implies that you also already should have a `state_map_type` and `assert_update_viable doesn`t need to know what `enkf_fs_type` is.

furthermore when you look at the only use of `assert_update_viable` which is `smoother_update` you see that `enkf_fs_type` is beeing passed then `state_map_type` is being fetch for validation. And later after the validation, yet again `state_map_type` is being fetch for other uses. 

simplification: validation function will only observe, and move the retrieval of `state_map_type` before the validation.

Reply:
Sounds good 👍